src/resource_manager/http_route.py

```diff
@@ -0,0 +1,213 @@
+            gateway_resource_information: GatewayResourceInformation state component.
+            http_route_type: Type of the HTTP route, can be http or https.
+        """
+        super().__init__(http_route_resource_information, gateway_resource_information)
+        self.http_route_type = http_route_type
+
+
+class HTTPRouteResourceManager(ResourceManager[GenericNamespacedResource]):
+    """service resource manager."""
+
+    def __init__(self, labels: LabelSelector, client: Client) -> None:
+        """Initialize the HTTPRouteResourceManager.
+
+        Args:
+            labels: Label to be added to created resources.
+            client: Initialized lightkube client.
+        """
+        self._client = client
+        self._labels = labels
+        self._http_route_generic_resource_class = create_namespaced_resource(
+            CUSTOM_RESOURCE_GROUP_NAME, "v1", HTTP_ROUTE_RESOURCE_NAME, HTTP_ROUTE_PLURAL
+        )
+
+    @map_k8s_auth_exception
+    def _gen_resource(self, resource_definition: ResourceDefinition) -> GenericNamespacedResource:
+        """Generate a Gateway resource from a gateway resource definition.
+
+        Args:
+            resource_definition: Part of charm state consisting of 2 components:
+                - HTTPRouteResourceInformation
+                - GatewayResourceDefinition
+
+        Returns:
+            A dictionary representing the gateway custom resource.
+        """
+        listener_id = (
+            f"{resource_definition.gateway_name}"
+            f"-{resource_definition.http_route_type.value}"
+            "-listener"
+        )
+        spec = {
+            "parentRefs": [
+                {
+                    "name": resource_definition.gateway_name,
+                    "namespace": self._client.namespace,
+                    "sectionName": listener_id,
+                }
+            ],
+        }
+        if resource_definition.http_route_type == HTTPRouteType.HTTPS:
+            spec["rules"] = [
+                {
+                    "matches": [
+                        {
+                            "path": {
+                                "type": "PathPrefix",
+                                "value": f"/{resource_definition.service_name}",
+                            }
+                        }
+                    ],
+                    "backendRefs": [
+                        {
+                            "name": resource_definition.service_name,
+                            "port": resource_definition.service_port,
+                        }
+                    ],
+                }
+            ]
+        else:
+            spec["rules"] = [
+                {
+                    "filters": [
+                        {
+                            "type": "RequestRedirect",
+                            "requestRedirect": {"scheme": "https", "statusCode": 301},
+                        }
+                    ]
+                }
+            ]
```

Comment:
Not sure this should be managed here, it's not a normal route but a "redirect" route. I'd probably create another resource manager named `HTTPtoHTTPSResourceManager` that inherits from `HTTPRouteResourceManager` and overrides _gen_resource

Reply:
makes sense, I created a `HTTPRouteRedirectResourceManager` class